# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextProviderCGL.mm
# Commit: 2b3e5d860458
# Full Hash: 2b3e5d860458cb3129b248b006ffe312ba52f3de
# Author: Markus Stange <mstange@themasta.com>
# Date: 2019-12-12 04:25:33
# Description:
#   Bug 1602813 - Stop calling -[NSOpenGLContext pixelFormat] because it's not available on 10.9. Instead, inline CreateWithFormat into this callsite. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56585
# ==============================================================================

diff -r e3362078c7bb -r 2b3e5d860458 gfx/gl/GLContextProviderCGL.mm
--- a/gfx/gl/GLContextProviderCGL.mm	Wed Dec 11 19:19:20 2019 +0000
+++ b/gfx/gl/GLContextProviderCGL.mm	Tue Dec 10 21:57:23 2019 +0000
@@ -145,12 +145,15 @@
 //     which GPU is currently driving the screen.
 static CGOpenGLDisplayMask GetFreshContextDisplayMask() {
   NSOpenGLPixelFormatAttribute attribs[] = {NSOpenGLPFAAllowOfflineRenderers, 0};
-  NSOpenGLContext* context = CreateWithFormat(attribs);
-  NSOpenGLPixelFormat* pixelFormat = [context pixelFormat];
+  NSOpenGLPixelFormat* pixelFormat = [[NSOpenGLPixelFormat alloc] initWithAttributes:attribs];
+  MOZ_RELEASE_ASSERT(pixelFormat);
+  NSOpenGLContext* context = [[NSOpenGLContext alloc] initWithFormat:pixelFormat
+                                                        shareContext:nullptr];
   GLint displayMask = 0;
   [pixelFormat getValues:&displayMask
             forAttribute:NSOpenGLPFAScreenMask
         forVirtualScreen:[context currentVirtualScreen]];
+  [pixelFormat release];
   [context release];
   return static_cast<CGOpenGLDisplayMask>(displayMask);
 }
